src/Lecture3.hs

```diff
@@ -213,8 +236,25 @@ types that can have such an instance.
 -- instance Foldable Weekday where
 -- instance Foldable Gold where
 -- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+
+instance Foldable List1 where
+  foldr :: (a -> b -> b) -> b -> List1 a -> b
+  foldr f acc (List1 x []) = f x acc
+  foldr f acc (List1 x xs) = f x (foldr f acc xs)
```

Comment:
You don't need to pattern match on two cases here, you can reuse the `Foldable` instance for lists 📂
It already handles the case of an empty list 🙂